Code/Framework/AzToolsFramework/Tests/EditorTransformComponentSelectionTests.cpp

```diff
@@ -188,6 +190,44 @@ namespace UnitTest
     ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
     // EditorTransformComponentSelection Tests
 
+    TEST_F(EditorTransformComponentSelectionFixture, Focus_is_not_changed_while_switching_viewport_interaction_request_instance)
+    {
+        // setup a dummy widget and make it the active window to ensure focus in/out events are fired
+        auto dummyWidget = AZStd::make_unique<QWidget>();
+        QApplication::setActiveWindow(dummyWidget.get());
+
+        // note: it is important to make sure the focus widget is parented to the dummy widget to have focus in/out events fire
+        auto focusWidget = AZStd::make_unique<UnitTest::FocusInteractionWidget>(dummyWidget.get());
+
+        // Given
+        // Setup viewport ui system
+        AzToolsFramework::ViewportUi::ViewportUiManager viewportUiManager;
+        viewportUiManager.ConnectViewportUiBus(AzToolsFramework::ViewportUi::DefaultViewportId);
+        viewportUiManager.InitializeViewportUi(&m_editorActions.m_defaultWidget, focusWidget.get());
+
+        // Begin EditorPickEntitySelection
+        using AzToolsFramework::EditorInteractionSystemViewportSelectionRequestBus;
+        EditorInteractionSystemViewportSelectionRequestBus::Event(
+            AzToolsFramework::GetEntityContextId(), &EditorInteractionSystemViewportSelectionRequestBus::Events::SetHandler,
+            [](const AzToolsFramework::EditorVisibleEntityDataCache* entityDataCache)
+            {
+                return AZStd::make_unique<AzToolsFramework::EditorPickEntitySelection>(entityDataCache);
+            });
+
+        // When
+        // A mouse event is sent to the focus widget (set to be the render overlay in the viewport ui system)
+        QTest::mouseClick(focusWidget.get(), Qt::MouseButton::LeftButton);
+
+        // Then
+        // focus should not change
+        EXPECT_FALSE(focusWidget->hasFocus());
```

Comment:
suggestion: We could cache and check that `QApplication::focusWidget` doesn't change to ensure that no other esoteric focus change behavior occurred and we're still on the correct widget

Reply:
Sure good shout I'll give that a try 👍